src/main/java/duke/program/TaskManager.java

```diff
@@ -0,0 +1,404 @@
+        System.out.println("Sorry bud, but your formatting is gibberish to me. Here is how it should be formatted: ");
+        System.out.println(command + " {description} /" + timeIndicatorCommand + " {timing}");
+    }
+
+    public static void printGenericErrorMessage() {
+        System.out.println("Oops! Something went wrong :(");
+    }
+
+    public void printMarkAsDoneMessage(Task task) {
+
+        String taskType = task.getType();
+        String taskStatus = task.getStatusIcon();
+        String taskDescription = task.getDescription();
+
+        System.out.println("Nice! I've marked this task as done:");
+        printIndividualTask(task, taskType, taskStatus, taskDescription);
+    }
+
+    public void printDeletedTaskMessage(Task task) {
+
+        String taskType = task.getType();
+        String taskStatus = task.getStatusIcon();
+        String taskDescription = task.getDescription();
+
+        System.out.println("Noted. I've removed this task:");
+        printIndividualTask(task, taskType, taskStatus, taskDescription);
+        System.out.println("Now you have " + (taskCount - 1) + " tasks in the list.");
+    }
+
+    private static void printIndividualTask(Task task, String taskType, String taskStatus, String taskDescription) {
+        if (taskType.equals(TASK_TYPE_ICON_TODO)) {
+            System.out.printf("[%s][%s] %s%n", taskType, taskStatus, taskDescription);
+        } else if (taskType.equals(TASK_TYPE_ICON_DEADLINE)) {
+            String taskByTime = task.getByDateTime();
+            System.out.printf("[%s][%s] %s (by: %s)%n", taskType, taskStatus, taskDescription, taskByTime);
+        } else if (taskType.equals(TASK_TYPE_ICON_EVENT)) {
+            String taskAtTime = task.getStartAndEndTime();
+            System.out.printf("[%s][%s] %s (at: %s)%n", taskType, taskStatus, taskDescription, taskAtTime);
+        } else {
+            printGenericErrorMessage();
+        }
+    }
+
+    public static void printTaskList(int taskCount, ArrayList<Task> tasks) {
+
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCount; i++) {
+
+            String taskType = tasks.get(i).getType();
+            String taskStatus = tasks.get(i).getStatusIcon();
+            String taskDescription = tasks.get(i).getDescription();
+
+            System.out.printf("%d.", i+1);
+            printIndividualTask(tasks.get(i), taskType, taskStatus, taskDescription);
+        }
+    }
+
+    public void addNewTodo(String rawDescription) throws EmptyDescriptionException {
+        String description = rawDescription.trim();
+        if (description.isEmpty()) {
+            throw new EmptyDescriptionException();
+        }
+        tasks.add(new ToDo(description));
+        taskCount++;
+    }
+
+    public void addNewDeadline(String descriptionAndTime) throws EmptyDescriptionException, InvalidFormatException {
```

Comment:
This line could exceed [>120 chars](https://se-education.org/guides/conventions/java/basic.html#layout), and affect readability. How about splitting into two lines?